File: DatingApp.API/Helpers/UserParams.cs
using DatingApp.API.Helpers.Pagination;

namespace DatingApp.API.Helpers
{
    public class UserParams : PaginationParams
    {
        public string CurrentUsername { get; set; }

        public string Gender { get; set; }

        public int MinAge { get; set; } = 18;

        public int MaxAge { get; set; } = 99;

        public string OrderBy { get; set; } = "lastActive";
    }

    public class MessageParams :  PaginationParams
    {
        public string Username { get; set; }

        public string Container { get; set; } = "Unread";
    }
}

File: client/src/models/userParams.ts
import { PaginationParams } from "./paginationParams";
import { User } from "./user";

export class UserParams extends PaginationParams {
  gender: string;
  minAge: number = 18;
  maxAge: number = 99;
  
  orderBy: string = 'lastActive';

  constructor(user: User) {
    super(5, 1);
    this.gender = user.gender === 'female' ? 'male' : 'female';
  }
}